WordPress/gutenberg
# Block Editor: LinkControl: Align documentation with current behavior

This pull request seeks to address a few consistency issues with the LinkControl documentation:

- There is no `className` prop
- `value` is not required
- `onChange` will always receive an item, never `null`
- The `onChange` example referenced the wrong (old) prop `onLinkChange`, and the example used an impossible scenario that the changed `item` could be falsey.
- The tests reference behavior of "required props", but there are no required props

Separately, I would like to improve documentation and types concerning what a value actually is. These are intended as an incremental step in that direction.

**Testing Instructions:**

The only changes impact documentation and tests. There is no expected impact on the behavior of the application runtime.

Ensure unit tests pass:

```
npm run test-unit
```
